backup-recovery.adoc

```diff
@@ -1,26 +1,101 @@
+
+If you want to migrate a particular onion address, you can replace the `btc_mainnet/tor/hiddenservice` folder in your data directory with the one from your backup.
+
+If you want to change your onion address, just delete the `btc_mainnet/tor/hiddenservice`. The next time you start Bisq, you'll have a different onion address.
+
+**Make sure you don't have any open offers, trades, or disputes before changing your onion address, as doing so will make you unreachable.**
+
+IMPORTANT: Before manually making any changes to Bisq data files, please remember to back up the current state of your Bisq data directory and make sure the Bisq program is closed.
+
+== Wallet Data
+
+Your wallet data is stored in `btc_mainnet/wallet`.
+
+Wallet issues (e.g., incorrect balance showing, corrupted/out-of-sync SPV file) can often be solved by resyncing the SPV chain and restarting the Bisq program. You can do this by clicking the `Delete SPV file and resync` button on the `Network info` tab of the `Settings` screen. If you can't access this screen, you can just delete the `bisq.spvchain` file in `btcmainnet/wallet` (just be advised that you will need to manually restart Bisq twice to complete the resync).
+
+If issues persist, you can try to restore a backup wallet file from `btc_mainnet/wallet/backup/backups_bisq_BTC_wallet`. This folder contains 20 of your most recent wallet backups (backups are made on a rolling basis). Replace the `bisq_BTC.wallet` wallet file in `btc_mainnet/wallet` with a backup wallet file and rename it to `bisq_BTC.wallet`. Start Bisq and see if your issue is fixed. If not, repeat with another backup.
+
+In extreme cases, if neither resyncing the SPV file nor restoring a wallet backup solves the problem, you might need to set up a new data directory. Make sure you don't have any open offers, trades, or disputes before doing this, as you won't be able to respond to these activities any longer.
+
+Here's how to do it:
+
+* make sure you have your Bisq wallet's <<secure-wallet#, seed words>>
+* close Bisq
+* delete the whole `Bisq` directory (after backing it up, just in case)
+* start Bisq (it will generate a fresh new data directory)
+* restore your wallet from seed words in the Bisq interface (on the `Wallet seed` tab of the `Account` screen)
+* optional: restore payment accounts and/or Tor keys through the processes outlined above
+
+**Make sure you don't have any open offers, trades, or disputes before doing this.**
+
+Please do not try to copy an old wallet backup folder (i.e., `btc_mainnet/wallet`) into a new data directory, as that will cause internal database files to be out of synx—just restore from seed words.
```

Comment:
The backup folders would not cause any issue. I think u can delete that sentence.